gapis/api/vulkan/templates/vk_spy_helpers.cpp.tmpl

```diff
@@ -420,18 +420,60 @@ uint32_t VulkanSpy::SpyOverride_vkCreateBuffer(
     }
 }
 
-uint32_t VulkanSpy::SpyOverride_vkCreateImage(
-        VkDevice                     device,
-        VkImageCreateInfo*           pCreateInfo,
-        VkAllocationCallbacks*       pAllocator,
-        VkImage*                     pImage) {
-    if (is_suspended() || is_observing()) {
-        VkImageCreateInfo override_create_info = *pCreateInfo;
-        override_create_info.musage |= VkImageUsageFlagBits::VK_IMAGE_USAGE_TRANSFER_SRC_BIT;
-        return  mImports.mVkDeviceFunctions[device].vkCreateImage(device, &override_create_info, pAllocator, pImage);
-    } else {
-        return  mImports.mVkDeviceFunctions[device].vkCreateImage(device, pCreateInfo, pAllocator, pImage);
+uint32_t VulkanSpy::SpyOverride_createImageAndCacheMemoryRequirements(
+    VkDevice device, VkImageCreateInfo* pCreateInfo,
+    VkAllocationCallbacks* pAllocator, VkImage* pImage,
+    VkMemoryRequirements* pMemoryRequirements) {
+  uint32_t result = VkResult::VK_SUCCESS;
+  VkImageCreateInfo override_create_info = *pCreateInfo;
+  override_create_info.musage |=
+      VkImageUsageFlagBits::VK_IMAGE_USAGE_TRANSFER_SRC_BIT;
+  result = mImports.mVkDeviceFunctions[device].vkCreateImage(
+      device, &override_create_info, pAllocator, pImage);
+  if (result != VkResult::VK_SUCCESS) {
+    return result;
+  }
+
+  if (pMemoryRequirements != nullptr) {
+    mImports.mVkDeviceFunctions[device].vkGetImageMemoryRequirements(
```

Comment:
This is only called in this override function, so swapchain images won't have valid memory requirement info populated. Their info will be all 0.